pytorch/ao
# move float8_experimental to torchao/float8

Summary:

This PR moves https://github.com/pytorch-labs/float8_experimental to torchao/float8.

There are no logic changes here. Here is how to reproduce this PR:
* copy float8_experimental/float8_experimental/* to torchao/float8
* copy float8_experimental/test/* to test/float8
* copy float8_experimental/benchmarks/* to benchmarks/float8
* copy the README over and delete sections which no longer apply (license, installation)
* replace `float8_experimental` with `torchao.float8` everywhere
* update tests to skip old PyTorch versions (only nightly supported)
* update non-emulated tests to require an H100 for now (we can enable CI in separate PR)
* update distributed tests to not run in CI for now (we can enable in separate PR)

Test Plan:

```
// run local tests, they pass
./test/float8/test_everything.sh

// run every benchmark in `benchmarks/float8`, they still work
```

Reviewers:

Subscribers:

Tasks:

Tags:
